File: src/model/exceptions/OccupiedPositionException.java
package model.exceptions;

import model.structs.Point;

public class OccupiedPositionException extends InvalidMoveException {
    public OccupiedPositionException(int x, int y) {
        super(String.format("Position (%d,%d) already occupied", x, y));
    }
    public OccupiedPositionException(Point point) {
        this(point.x, point.y);
    }
}

File: src/model/exceptions/OccupiedPositionException.kt
package model.exceptions

import model.structs.Point

class OccupiedPositionException(x: Int, y: Int) :
    InvalidMoveException(String.format("Position (%d,%d) already occupied", x, y)) {
    constructor(point: Point) : this(point.x, point.y)
}
